services/movie_session.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(movie_show_time: str,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(
+            show_time__date=datetime.strptime(session_date, "%Y-%m-%d")
+        )
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: datetime = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> QuerySet:
+    queryset = MovieSession.objects.filter(id=session_id)
+
+    if show_time is not None:
+        queryset.update(show_time=show_time)
+
+    if movie_id is not None:
+        queryset.update(movie_id=movie_id)
+
+    if cinema_hall_id is not None:
+        queryset.update(cinema_hall_id=cinema_hall_id)
```

Comment:
You do 3 queries. It is not okay. You have 2 options:
1. Get the object, set all new values, and then save it(2 queries)
2. Create and form a dictionary with fields to update and then do .update(1 query)